engine/executor/dm/unitholder_test.go

```diff
@@ -124,6 +125,47 @@ func TestUnitHolder(t *testing.T) {
 	require.NoError(t, unitHolder.Resume(context.Background()))
 }
 
+func TestUnitHolderBinlog(t *testing.T) {
+	unitHolder := &unitHolderImpl{}
+	unitHolder.unit = &dumpling.Dumpling{}
+
+	// wrong type
+	msg, err := unitHolder.Binlog(context.Background(), &dmpkg.BinlogTaskRequest{})
+	require.Error(t, err)
+	require.Equal(t, "", msg)
+	// no binlog error
+	unitHolder.unit = syncer.NewSyncer(&config.SubTaskConfig{Flavor: mysql.MySQLFlavor}, nil, nil)
+	unitHolder.runCtx = context.Background()
+	msg, err = unitHolder.Binlog(context.Background(), &dmpkg.BinlogTaskRequest{})
+	require.EqualError(t, err, "source '' has no error")
+	require.Equal(t, "", msg)
+	// binlog skip
+	msg, err = unitHolder.Binlog(context.Background(), &dmpkg.BinlogTaskRequest{Op: pb.ErrorOp_Skip, BinlogPos: "mysql-bin.000001:2345"})
+	require.Nil(t, err)
+	require.Equal(t, "", msg)
+}
+
+func TestUnitHolderBinlogSchema(t *testing.T) {
+	unitHolder := &unitHolderImpl{}
+	unitHolder.unit = &dumpling.Dumpling{}
+
+	// wrong type
+	msg, err := unitHolder.BinlogSchema(context.Background(), &dmpkg.BinlogSchemaTaskRequest{})
+	require.Error(t, err)
+	require.Equal(t, "", msg)
+	// wrong stage
+	unitHolder.unit = syncer.NewSyncer(&config.SubTaskConfig{Flavor: mysql.MySQLFlavor}, nil, nil)
+	unitHolder.runCtx = context.Background()
+	msg, err = unitHolder.BinlogSchema(context.Background(), &dmpkg.BinlogSchemaTaskRequest{})
+	require.EqualError(t, err, fmt.Sprintf("current stage is %d but not paused, invalid", metadata.StageRunning))
```

Comment:
maybe we can add String for the stage later.

And message can be "invalid request, current stage is %d, expected stage is paused"